crates/ruff_python_semantic/src/scope.rs

```diff
@@ -27,6 +27,9 @@ pub struct Scope<'a> {
     /// A map from bound name to binding ID.
     bindings: FxHashMap<&'a str, BindingId>,
 
+    /// A map from binding ID to bound name.
+    binding_name: FxHashMap<BindingId, &'a str>,
+
```

Comment:
Hash maps are kind of expensive (48 bytes or something in size and require an allocation per scope). Could we instead move the name to `Binding`, recompute it on demand (implement a helper on `binding` that uses `BindingId` to extract the name, I would prefer this approach), or store it as a another `IndexVec<BindingId, &'str>` on the semantic model (since we need an entry for **every** name).